tests/flows/test_fre_flow.py

```diff
@@ -0,0 +1,130 @@
+#
+# Copyright 2021 Ocean Protocol Foundation
+# SPDX-License-Identifier: Apache-2.0
+#
+import os
+
+from ocean_lib.example_config import ExampleConfig
+from ocean_lib.models.erc20_token import ERC20Token
+from ocean_lib.models.erc721_factory import ERC721FactoryContract
+from ocean_lib.models.fixed_rate_exchange import FixedRateExchange
+from ocean_lib.models.models_structures import ErcCreateData, FixedData
+from ocean_lib.ocean.mint_fake_ocean import mint_fake_OCEAN
+from ocean_lib.ocean.ocean import Ocean
+from ocean_lib.ocean.util import get_address_of_type
+from ocean_lib.web3_internal.constants import ZERO_ADDRESS
+from ocean_lib.web3_internal.currency import to_wei
+from ocean_lib.web3_internal.wallet import Wallet
+
+
+def test_fre_flow():
+    config = ExampleConfig.get_config()
+    ocean = Ocean(config)
+
+    # Create Alice's wallet
+    alice_private_key = os.getenv("TEST_PRIVATE_KEY1")
+    alice_wallet = Wallet(
+        ocean.web3,
+        alice_private_key,
+        config.block_confirmations,
+        config.transaction_timeout,
+    )
+    assert alice_wallet.address
+
+    # Mint OCEAN
+    mint_fake_OCEAN(config)
+    assert alice_wallet.web3.eth.get_balance(alice_wallet.address) > 0, "need ETH"
+
+    # Publish an NFT token
+    nft_token = ocean.create_nft_token(
+        "NFTToken1", "NFT1", alice_wallet, "https://oceanprotocol.com/nft/"
+    )
+    token_address = nft_token.address
+    assert token_address
+
+    # Prepare data for ERC20 token
+    erc20_data = ErcCreateData(
+        template_index=1,
+        strings=["Datatoken 1", "DT1"],
+        addresses=[
+            alice_wallet.address,
+            alice_wallet.address,
+            ZERO_ADDRESS,
+            ocean.OCEAN_address,
+        ],
+        uints=[ocean.web3.toWei(200, "ether"), 0],
+        bytess=[b""],
+    )
+
+    tx_id = nft_token.create_erc20(erc20_data, alice_wallet)
+    tx_receipt = ocean.web3.eth.wait_for_transaction_receipt(tx_id)
+
+    erc721_factory = ERC721FactoryContract(
+        ocean.web3, get_address_of_type(config, "ERC721Factory")
+    )
+
+    registered_event = erc721_factory.get_event_log(
+        ERC721FactoryContract.EVENT_TOKEN_CREATED,
+        tx_receipt.blockNumber,
+        ocean.web3.eth.block_number,
+        None,
+    )
+
+    erc20_address = registered_event[0].args.newTokenAddress
+    print(f"token_address = '{erc20_address}'")
+
+    # Mint the datatokens
+    fixed_price_address = get_address_of_type(config, "FixedPrice")
```

Comment:
I would rather use get_address_of_type only inside the Ocean class. You can reuse the function that GitHub hides from me (ocean.get_fre_contract...)